overlord/snapstate/handlers.go

```diff
@@ -1690,6 +1815,8 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 	}
 	if snapsup.MigratedToExposedHome {
 		snapst.MigratedToExposedHome = true
+	} else if snapsup.DisableHomeMigration {
+		snapst.MigratedToExposedHome = false
```

Comment:
so we don't need to undo this in undoLinkSnap because the snap is unaway unliked? so we take care of this only when we reach undoUnlinkCurrenSnap?